src/station_cmd.cpp

```diff
@@ -671,14 +671,12 @@ static void UpdateStationSignCoord(BaseStation *st)
  */
 static CommandCost BuildStationPart(Station **st, DoCommandFlag flags, bool reuse, TileArea area, StationNaming name_class)
 {
+	assert(*st == NULL || (*st)->owner == _current_company);
```

Comment:
I think this assertion can be triggered by a malicious client to crash the server.
When distant-joining an existing station, the client sents the station id of the station to join.
At this point the id has been checked for existence, but not for ownership.

So, this must not be an "assert(...)" but a "if (..) return CMD_ERROR;"

Reply:
Good thinking, looks like that could be the case indeed. Fixed it.